### PR TITLE
iox-#27 integration tests

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

This PR adds integration tests for client and server and fixed an issue found with the tests.

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Relates to #27

### CODE REVIEW

File: iceoryx_posh/source/roudi/port_manager.cpp [hunk 1]
```diff
@@ -1063,6 +1067,7 @@ void PortManager::removePublisherFromServiceRegistry(const capro::ServiceDescrip
 
 void PortManager::addServerToServiceRegistry(const capro::ServiceDescription& service) noexcept
 {
+    std::cout << "#### " << service << std::endl;
```
Comment (elfenpiff) [inline iceoryx_posh/source/roudi/port_manager.cpp:1070]: Is this some kind of printf debugging artifact ;)

File: iceoryx_posh/test/integrationtests/test_client_server.cpp [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+  public:
+    void SetUp() override
+    {
+        PoshRuntime::initRuntime("ua");
+        deadlockWatchdog.watchAndActOnFailure([] { std::terminate(); });
+    }
+
+    static constexpr iox::units::Duration DEADLOCK_TIMEOUT{5_s};
+    Watchdog deadlockWatchdog{DEADLOCK_TIMEOUT};
+    ServiceDescription sd{"blue", "and", "yellow"};
```
Comment (elfenpiff) [inline iceoryx_posh/test/integrationtests/test_client_server.cpp:70]: I think hypnotoad likes this.

File: iceoryx_posh/test/integrationtests/test_client_server.cpp [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+    }
+    uint64_t sum{0U};
+};
+
+class ClientServer_test : public RouDi_GTest
+{
+  public:
+    void SetUp() override
+    {
+        PoshRuntime::initRuntime("ua");
```
Comment (elfenpiff) [inline iceoryx_posh/test/integrationtests/test_client_server.cpp:64]: Please add more lyrical esprit (optional)

File: iceoryx_posh/source/roudi/port_manager.cpp (hunk 1 above)
Comment (elBoberido) [reply iceoryx_posh/source/roudi/port_manager.cpp:1070]: upsi

File: iceoryx_posh/test/integrationtests/test_client_server.cpp [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+    constexpr int64_t SEQUENCE_ID{73};
+    constexpr uint64_t AUGEND{13U};
+    constexpr uint64_t ADDEND{42U};
+
+    Client<DummyRequest, DummyResponse> client{sd};
+    Server<DummyRequest, DummyResponse> server{sd};
+
+    // send request
+    {
+        auto loanResutl = client.loan();
```
Comment (mossmaurice) [inline iceoryx_posh/test/integrationtests/test_client_server.cpp:90]:
```suggestion
        auto loanResult = client.loan();
```

File: iceoryx_posh/test/integrationtests/test_client_server.cpp [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+
+using namespace iox::popo;
+using namespace iox::capro;
+using namespace iox::runtime;
+
+class DummyRequest
+{
+  public:
+    DummyRequest() = default;
+    DummyRequest(uint64_t augend, uint64_t addend)
```
Comment (mossmaurice) [inline iceoryx_posh/test/integrationtests/test_client_server.cpp:39]: Why `augend`?

File: iceoryx_posh/test/integrationtests/test_client_server.cpp [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+    constexpr int64_t SEQUENCE_ID{37};
+    constexpr uint64_t AUGEND{7U};
+    constexpr uint64_t ADDEND{4U};
+
+    UntypedClient client{sd};
+    UntypedServer server{sd};
+
+    // send request
+    {
+        auto loanResutl = client.loan(sizeof(DummyRequest), alignof(DummyRequest));
```
Comment (mossmaurice) [inline iceoryx_posh/test/integrationtests/test_client_server.cpp:135]:
```suggestion
        auto loanResult = client.loan(sizeof(DummyRequest), alignof(DummyRequest));
```

File: iceoryx_posh/test/integrationtests/test_client_server.cpp [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+    Client<DummyRequest, DummyResponse> client_1{sd};
+    Client<DummyRequest, DummyResponse> client_2{sd};
+    std::vector<Client<DummyRequest, DummyResponse>*> client{&client_1, &client_2};
+
+    Server<DummyRequest, DummyResponse> server{sd};
+
+    // send requests
+    for (auto i = 0U; i < NUMBER_OF_REQUESTS; ++i)
+    {
+        auto loanResutl = client[i]->loan(AUGEND[i], ADDEND[i]);
```
Comment (mossmaurice) [inline iceoryx_posh/test/integrationtests/test_client_server.cpp:189]:
```suggestion
        auto loanResult = client[i]->loan(AUGEND[i], ADDEND[i]);
```

File: iceoryx_posh/test/moduletests/test_roudi_portmanager_client_server.cpp [hunk 8]
```diff
@@ -491,6 +491,84 @@ TEST_F(PortManager_test, ServerStateIsForwardedToInterfacePortWhenAlreadyOfferAn
 
 // END forwarding to InterfacePort tests
 
+// BEGIN service registry tests
```
Comment (mossmaurice) [inline iceoryx_posh/test/moduletests/test_roudi_portmanager_client_server.cpp:494]: Should those tests cases not be added to a different `.cpp`?

**Review by mossmaurice — COMMENTED** [review]
LGTM, few nitpicks.

File: iceoryx_posh/test/moduletests/test_roudi_portmanager_client_server.cpp (hunk 8 above)
Comment (elBoberido) [reply iceoryx_posh/test/moduletests/test_roudi_portmanager_client_server.cpp:494]: Well, it's part of the tests to check whether the PortManager adds the ServerPort to the ServiceRegistry. For me this belongs together.

File: iceoryx_posh/test/integrationtests/test_client_server.cpp (hunk 5 above)
Comment (elBoberido) [reply iceoryx_posh/test/integrationtests/test_client_server.cpp:39]: https://en.wikipedia.org/wiki/Addition#Notation_and_terminology :smirk:

File: iceoryx_posh/test/moduletests/test_roudi_portmanager_client_server.cpp (hunk 8 above)
Comment (elBoberido) [reply iceoryx_posh/test/moduletests/test_roudi_portmanager_client_server.cpp:494]: ... maybe the naming of the comment is poor?

Comment (mossmaurice) [reply iceoryx_posh/test/moduletests/test_roudi_portmanager_client_server.cpp:494]: What about "BEGIN PortManager interaction tests with service registry"?

Comment (elBoberido) [reply iceoryx_posh/test/moduletests/test_roudi_portmanager_client_server.cpp:494]: Sure. The question is just if we want to have another commit and CI run for the change of the comment? I think the fact that the test is int the `PortManager` already shows that it is related to the port manager. It's up to you, if you want I do a commit.

Comment (mossmaurice) [reply iceoryx_posh/test/moduletests/test_roudi_portmanager_client_server.cpp:494]: Ok, for me to stick to with this comment.

**Review by mossmaurice — APPROVED** [review]

**Review by elfenpiff — APPROVED** [review]

**Pull request merged**